src/figuresClasses.ts

```diff
@@ -1,11 +1,84 @@
-export interface Figure {}
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('your error message');
+    }
+
+    const arrOfSides = [a, b, c];
+    const maxSide = Math.max(...arrOfSides);
+    const smallerSidesSum = arrOfSides
+      .filter((item) => item < maxSide)
+      .reduce((acc, currentValue) => acc + currentValue);
+
+    if (maxSide >= smallerSidesSum) {
+      throw new Error('your error message');
+    }
+  }
+
+  public shape = 'triangle';
+
+  public getArea(): number {
+    const a = this.a;
+    const b = this.b;
+    const c = this.c;
+
+    const p = (a + b + c) / 2;
+
+    const area = p * (p - a) * (p - b) * (p - c);
```

Comment:
The calculation of the area for the triangle uses Heron's formula, but the variable 'area' should be calculated as 'Math.sqrt(p * (p - a) * (p - b) * (p - c))'. Currently, it calculates the product without taking the square root, which is incorrect.